### ISSUE TITLE
predictions = model(inputs) outputs nan

### ISSUE THREAD

**Issue by deboss2020** [tensorflow/tensorflow#46690]
Hi

This is a great project and is very useful, but I have encountered an error with model predicting

I have used an lstm as the model input. Once I run window.plot(model=lstm_model) the output of predictions is just a 2D array with "nan" as each value. The shape of this array is 2, 79, 1

Please help

P.S. Here's my code for reference

print("Initiating the time series forcasting module")

print("Importing modules")
import pandas as pd
import os
import matplotlib as mpl
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns
import tensorflow as tf

print("Collecting data")
df = pd.read_csv("data.csv")
dates = pd.to_datetime((df.pop("Date")), format = "%Y/%m/%d")

print("Splitting data")
column_indices = {name: i for i, name in enumerate(df.columns)}
n = len(df)
train_df = df[0:int(n*0.9)]
test_df = df[int(n*0.9):]
print(str(len(train_df)) + " plots for training")
print(str(len(test_df)) + " plots for testing")
num_features = df.shape[1]

print("Normalizing data")
train_mean = train_df.mean()
train_std = train_df.std()
train_df = (train_df - train_mean) / train_std
test_df = (test_df - train_mean) / train_std
df_std = (df - train_mean) / train_std

print("Creating classes")
class WindowGenerator():
    def __init__(self, input_width, label_width, shift,
                label_columns=None, train_df=train_df, test_df=test_df):
        print("Generating windows")
        self.train_df = train_df
        self.test_df = test_df
        
        self.label_columns = label_columns
        if label_columns is not None:
            self.label_column_indices = {name: i for i, name in
                                         enumerate(label_columns)}
        self.column_indices = {name: i for i, name in
                               enumerate(train_df.columns)}
        self.input_width = input_width
        self.label_width = label_width
        self.shift = shift
        
        self.total_window_size = input_width + shift
        self.input_slice = slice(0, input_width)
        self.input_indices = np.arange(self.total_window_size)[self.input_slice]
        
        self.label_start = self.total_window_size - self.label_width
        self.labels_slice = slice(self.label_start, None)
        self.label_indices = np.arange(self.total_window_size)[self.labels_slice]
    
    def __repr__(self):
        return "\n".join(["Total window size: " + str(self.total_window_size),
                          "Input indices: " + str(self.input_indices),
                          "Label indices: " + str(self.label_indices),
                          "Label column names: " + str(self.label_columns)])
    
    def split_window(self, features):
        print("Splitting windows")
        inputs = features[:, self.input_slice, :]
        labels = features[:, self.labels_slice, :]
        if self.label_columns is not None:
            labels = tf.stack(
                [labels[:, :, self.column_indices[name]] for name in self.label_columns],
                axis=-1)
        inputs.set_shape([None, self.input_width, None])
        labels.set_shape([None, self.label_width, None])
        return inputs, labels
    
    def plot(self, plot_col, model=None, max_subplots=3):
        print("Plotting data")
        inputs, labels = self.example
        plt.figure(figsize=(12,8))
        plot_col_index = self.column_indices[plot_col]
        max_n = min(max_subplots, len(inputs))
        for n in range(max_n):
            plt.subplot(3,1,n+1)
            plt.ylabel(plot_col)
            plt.plot(self.input_indices, inputs[n, :, plot_col_index],
                     label="Inputs", marker=".", zorder=-10)
            if self.label_columns:
                label_col_index = self.label_column_indices.get(plot_col, None)
            else:
                label_col_index = plot_col_index
                
            if plot_col_index is None:
                continue
            
            plt.scatter(self.label_indices, labels[n, :, label_col_index],
                        edgecolors="k", label="Labels", c='#2ca02c', s=64)
            
            if model is not None:
                predictions = model(inputs)
                plt.scatter(self.label_indices, predictions[n, :, label_col_index],
                    marker='X', edgecolors='k', label='Predictions',
                    c='#ff7f0e', s=64)
            if n == 0:
                plt.legend()
        plt.xlabel("Date")
        plt.show()
    
    def make_ds(self,data):
        print("Generating dataset")
        data = np.array(data, dtype=np.float32)
        ds = tf.keras.preprocessing.timeseries_dataset_from_array(
            data=data,
            targets=None,
            sequence_length=self.total_window_size,
            sequence_stride=1,
            shuffle=True,
            batch_size=32,)
        ds = ds.map(self.split_window)
        return ds
    
    @property
    def train(self):
        return self.make_ds(self.train_df)
    
    @property
    def test(self):
        return self.make_ds(self.test_df)
    
    @property
    def example(self):
        result = getattr(self, "_example", None)
        if result is None:
            result = next(iter(self.train))
            self._example = result
        return result
    
class Baseline(tf.keras.Model):
  def __init__(self, label_index=None):
    print("Creating baseline model")
    super().__init__()
    self.label_index = label_index

  def call(self, inputs):
    if self.label_index is None:
      return inputs
    result = inputs[:, :, self.label_index]
    return result[:, :, tf.newaxis]
    
wide_window = WindowGenerator(
    input_width=24, label_width=24, shift=24,
    label_columns=['ConfirmedCases'])

def compile_and_fit(model, window, patience=2, MAX_EPOCHS=20):
  model.compile(loss=tf.losses.MeanSquaredError(),
                optimizer=tf.optimizers.Adam(),
                metrics=[tf.metrics.MeanAbsoluteError()])

  history = model.fit(window.train, epochs=MAX_EPOCHS)
  return history

lstm_model = tf.keras.models.Sequential([
    tf.keras.layers.LSTM(32, return_sequences=True),
    tf.keras.layers.Dense(units=1)
])

history = compile_and_fit(lstm_model, wide_window)

wide_window.plot(model=lstm_model, plot_col="ConfirmedCases")

**Comment by Saduf2019** [comment]
@deboss2020 
We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or stand alone code to reproduce the issue faced if not shared]

**Comment by Saduf2019** [comment]
@deboss2020 
Please share a colab gist of the error reported.
